app/Http/Controllers/BookController.php

```diff
@@ -15,11 +16,23 @@ class BookController extends Controller
     public function __construct(Book $book) 
     {
         $this->book=$book;
+
     }
 
-    public function getAllBooks()
-    {
+    public function getAllBooks(Request $request)
+    {     
+       
         $books = $this->book->all();
+       
+        if ($request->has('genre')) {
+            $genre = Genre::where('name', $request->genre)->first();
+    
+            if ($genre) {
+                $books = $books->where('genre_id', $genre->id);
+            } else {
+                return response()->json(['message' => 'Genre not found'], 404);
+            }
+        }
```

Comment:
What you've done here is really good, but as per the documentation the way these genre filter should work is by the id of the genre, not the name. It should be like `/api/books?genre=1` for example.

[https://github.com/iO-Academy/book-swap-api-docs#return-all-books---optionally-filtered](https://github.com/iO-Academy/book-swap-api-docs#return-all-books---optionally-filtered)

If it weren't for the requirements in the docs, I would be very happy to accept what you've done as a perfectly valid way of filtering.

Reply:
Once you have done that, a nice way of making sure the genre is valid is by using the `exists` option for the $request->validate() function

For example you can do: `exists:genres,id`

This will automatically check the database to make sure the value set for ?genre exists in the genres table's id column